pyctcdecode/alphabet.py

```diff
@@ -3,37 +3,74 @@
 from typing import List, Optional
 
 
+UNK_CHAR = "⁇"  # representation of unknown character in regular alphabet
 BPE_CHAR = "▁"  # character used for token boundary if BPE is used
 UNK_BPE_CHAR = "▁⁇▁"  # representation of unknown character in BPE
 
 logger = logging.getLogger(__name__)
 
 
-def _get_ctc_index(label_list: List[str]) -> int:
-    """Get index of ctc blank character in alphabet."""
-    return len(label_list) - 1 if label_list[-1] == "" else -1
-
-
-def _normalize_alphabet(label_list: List[str], ctc_token_idx: Optional[int] = None) -> List[str]:
-    """Normalize alphabet for non-bpe decoder."""
-    if any([len(c) > 1 for c in label_list]):
-        raise ValueError("For non-bpe alphabet only length 1 entries and blank token are allowed.")
-    if ctc_token_idx is None:
-        ctc_token_idx = _get_ctc_index(label_list)
-    clean_labels = label_list[:]
-    # check for space token
-    if " " not in clean_labels:
-        raise ValueError("Space token ' ' missing from vocabulary.")
-    # specify ctc blank token
-    if ctc_token_idx == -1:
-        clean_labels.append("")
+def _check_if_bpe(labels: List[str]):
+    """Check if input alphabet is BPE or not."""
+    if any([s.startswith("##") for s in labels]) or any([s.startswith(BPE_CHAR) for s in labels]):
+        logger.info("Alphabet determined to be of BPE style.")
+        is_bpe = True
     else:
-        clean_labels[ctc_token_idx] = ""
-    return clean_labels
+        logger.info("Alphabet determined to be of regular style.")
+        is_bpe = False
+    return is_bpe
+
+
+def _normalize_regular_alphabet(labels: List[str]) -> List[str]:
```

Comment:
yeah this method is the big addition in this diff to add a lot of convenience to support different sorts of alphabets without the user having to think about. Since it's heavy in what it tries to do we are adding lots of logging. I can try to tease it apart a bit into more sane functional parts, don't not sure if there is a great way to do this via validators